src/vm/functions/database.rs

```diff
@@ -4,18 +4,20 @@ use vm::functions::tuples;
 use vm::functions::tuples::TupleDefinitionType::{Implicit, Explicit};
 
 use vm::types::{Value, OptionalData, BuffData, PrincipalData, BlockInfoProperty, TypeSignature};
-use vm::representations::{SymbolicExpression};
-use vm::errors::{CheckErrors, InterpreterError, RuntimeErrorType, InterpreterResult as Result,
-                 check_argument_count, check_arguments_at_least};
+use vm::representations::{SymbolicExpression, SymbolicExpressionType};
+use vm::errors::{CheckErrors, InterpreterError, RuntimeErrorType, InterpreterResult as Result, check_argument_count, check_arguments_at_least};
 use vm::{eval, LocalContext, Environment};
 
 pub fn special_contract_call(args: &[SymbolicExpression],
                              env: &mut Environment,
                              context: &LocalContext) -> Result<Value> {
     check_arguments_at_least(2, args)?;
 
-    let contract_name = args[0].match_atom()
-        .ok_or(CheckErrors::ExpectedName)?;
+    let contract_identifier = match args[0].expr {
+        SymbolicExpressionType::AtomValue(Value::Principal(PrincipalData::Contract(ref contract_identifier))) => contract_identifier,
```

Comment:
Like elsewhere, this should error on `AtomValue`.

Reply:
Addressed with #c871057